### ISSUE TITLE
Incorrect `is` behaviour with NewType

### ISSUE THREAD

**Issue by decorator-factory** [microsoft/pyright#11463]
## Bug
Pyright considers `left is right` to always be false in cases where `left` is a `NewType` instance with a base of `T` and `right` is a `T` literal.

## Code
```python
from types import NoneType
from typing import NewType, reveal_type

Apple = NewType("Apple", NoneType)
Banana = NewType("Banana", bool)
Cherry = NewType("Cherry", int)

def f(a: Apple, b: Banana, c: Cherry) -> None:
    if a is None:
        reveal_type(a)  # pyright reports unreachable

    if b is True:
        reveal_type(b)  # pyright reports unreachable

    if c is False:
        reveal_type(c)  # pyright reports unreachable

f(Apple(None), Banana(True), Cherry(False))
```

Runtime output:
```
Runtime type is 'NoneType'
Runtime type is 'bool'
Runtime type is 'bool'
```

Interestingly, pyright also doesn't produce `reportUnnecessaryComparison` diagnostics for any of the conditions.

(this also works with `int` and an integer literal, but `is` with an integer literal is bad so I didn't include it)

## Pyright version
pyright 1.1.409, python 3.14

**Comment by decorator-factory** [comment]
Possibly related:
```py
from typing import NewType, reveal_type

class Fruit: pass

Apple = NewType("Apple", Fruit)
Orange = NewType("Orange", Fruit)

def f(a: Apple, b: Orange) -> None:
    if a is b:
    #  ^Condition will always evaluate to False since the types "Apple" and "Orange" have no overlap
    #  (reportUnnecessaryComparison)
        reveal_type(a)
        # ^However, this code is considered reachable, and reveals the type as "Apple"

fruit = Fruit()
f(Apple(fruit), Orange(fruit))
```
It's probably good to emit some kind of diagnostic, since comparing different `NewType`s might be a mistake that `NewType` was designed to prevent in the first place, but it's definitely possible for this condition to be True at runtime.
